manifests/update.pp

```diff
@@ -56,13 +56,27 @@
   } else {
     $_refresh = true
   }
+  # We perform the update in an `unless` clause of the exec, and
+  # return true only if the package cache file changed.
+  # This ensures that Puppet does not report a change if the
+  # update command had no effect. See MODULES-10763 for discussion.
+  $apt_update_had_no_effect = epp(
+    "${module_name}/update_had_no_effect.sh.epp",
+    {
+      'provider' => $apt::provider,
+      'timeout'  => $apt::_update['timeout'],
+      'tries'    => $apt::_update['tries'],
+    }
+  )
   exec { 'apt_update':
-    command     => "${apt::provider} update",
+    command     => "echo ${apt::provider} successfully updated the package cache.",
```

Comment:
I think we wanted to execute the command here but we are prefixing with echo which means we do nothing except echo which is conflicting with original design.
it should as it is.
```
command => "${apt::provider} update",
```
This will get translate to `apt update` or `dnf update` based on the type of provider.

Reply:
@Ramesh7 

You are missing the point.  We want the main exec to be an echo reporting that the package cache was updated.

Actually, a `notice` might be more appropriate.

It's the `unless` clause down below that does the actual work.